core/engine/src/builtins/temporal/plain_month_day/mod.rs

```diff
@@ -128,3 +278,50 @@ pub(crate) fn create_temporal_month_day(
     // 9. Return object.
     Ok(obj.into())
 }
+
+fn to_temporal_month_day(
+    item: &JsValue,
+    options: &JsObject,
+    context: &mut Context,
+) -> JsResult<JsValue> {
+    let overflow = get_option::<ArithmeticOverflow>(options, js_str!("overflow"), context)?
+        .unwrap_or(ArithmeticOverflow::Constrain);
+
+    // get the calendar property (string) from the item object
+    let calender_id = item.get_v(js_str!("calendar"), context)?;
+    let calendar = to_temporal_calendar_slot_value(&calender_id)?;
+
+    let inner = if item.is_object() {
+        if let Some(data) = item
+            .as_object()
+            .and_then(JsObject::downcast_ref::<PlainMonthDay>)
+        {
+            data.inner.clone()
+        } else {
+            InnerMonthDay::new(
+                item.get_v(js_str!("month"), context)
```

Comment:
`to_integer_with_truncation` for `temporal/mod.rs` should be usable when getting the items value that way another `expect` can be removed.

For instance, something akin to the `PlainDate` [constructor](https://github.com/boa-dev/boa/blob/main/core/engine/src/builtins/temporal/plain_date/mod.rs#L254)